### PR TITLE
KDF documentation updates.

### PR DESCRIPTION
**Pull request by paulidale**
Remove references to the control functions and replace them with the set param function.
Remove references to the new by ID function and use the new by name instead.

- [x] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

File: doc/man7/EVP_KDF-HKDF.pod [hunk 1]
```diff
@@ -135,16 +135,16 @@ RFC 5869
...
 
 L<EVP_KDF>,
-L<EVP_KDF-CTX_new_id(3)>,
+L<EVP_KDF-CTX_new(3)>,
 L<EVP_KDF-CTX_free(3)>,
-L<EVP_KDF-ctrl(3)>,
 L<EVP_KDF-size(3)>,
+L<EVP_KDF-CTX_set_params(3)>,
 L<EVP_KDF-derive(3)>,
 L<EVP_KDF-CTX(3)/PARAMETERS>
```
Comment (levitte) [inline doc/man7/EVP_KDF-HKDF.pod]: These links are to actual functions. How did they get dashes?

File: doc/man7/EVP_KDF-PBKDF2.pod [hunk 2]
```diff
@@ -82,9 +82,9 @@ SP800-132
 =head1 SEE ALSO
 
 L<EVP_KDF>,
-L<EVP_KDF-CTX_new_id(3)>,
+L<EVP_KDF-CTX_new(3)>,
 L<EVP_KDF-CTX_free(3)>,
-L<EVP_KDF-ctrl(3)>,
+L<EVP_KDF-CTX_set_params(3)>,
 L<EVP_KDF-derive(3)>,
 L<EVP_KDF-CTX(3)/PARAMETERS>
```
Comment (levitte) [inline doc/man7/EVP_KDF-PBKDF2.pod]: Same thing here

File: doc/man7/EVP_KDF-SCRYPT.pod [hunk 3]
```diff
@@ -124,15 +124,15 @@ RFC 7914
 =head1 SEE ALSO
 
 L<EVP_KDF>,
-L<EVP_KDF-CTX_new_id(3)>,
+L<EVP_KDF-CTX_new(3)>,
 L<EVP_KDF-CTX_free(3)>,
-L<EVP_KDF-ctrl(3)>,
+L<EVP_KDF-CTX_set_params(3)>,
 L<EVP_KDF-derive(3)>,
 L<EVP_KDF-CTX(3)/PARAMETERS>
```
Comment (levitte) [inline doc/man7/EVP_KDF-SCRYPT.pod]: Same here

File: doc/man7/EVP_KDF-SS.pod [hunk 4]
```diff
@@ -173,9 +173,9 @@ NIST SP800-56Cr1.
...
 
 L<EVP_KDF>,
-L<EVP_KDF-CTX_new_id(3)>,
+L<EVP_KDF-CTX_new(3)>,
 L<EVP_KDF-CTX_free(3)>,
-L<EVP_KDF-ctrl(3)>,
+L<EVP_KDF-CTX_set_params(3)>,
 L<EVP_KDF-size(3)>,
 L<EVP_KDF-derive(3)>,
 L<EVP_KDF(3)/PARAMETERS>
```
Comment (levitte) [inline doc/man7/EVP_KDF-SS.pod]: Same here

File: doc/man7/EVP_KDF-SSHKDF.pod [hunk 5]
```diff
@@ -139,16 +139,16 @@ RFC 4253
...
 
 L<EVP_KDF>,
-L<EVP_KDF-CTX_new_id(3)>,
+L<EVP_KDF-CTX_new(3)>,
 L<EVP_KDF-CTX_free(3)>,
-L<EVP_KDF-ctrl(3)>,
+L<EVP_KDF-CTX_set_params(3)>,
 L<EVP_KDF-size(3)>,
 L<EVP_KDF-derive(3)>,
 L<EVP_KDF(3)/PARAMETERS>
```
Comment (levitte) [inline doc/man7/EVP_KDF-SSHKDF.pod]: Same here

File: doc/man7/EVP_KDF-TLS1_PRF.pod [hunk 6]
```diff
@@ -95,9 +95,9 @@ RFC 2246, RFC 5246 and NIST SP 800-135 r1
 =head1 SEE ALSO
 
 L<EVP_KDF>,
-L<EVP_KDF-CTX_new_id(3)>,
+L<EVP_KDF-CTX_new(3)>,
 L<EVP_KDF-CTX_free(3)>,
-L<EVP_KDF-ctrl(3)>,
+L<EVP_KDF-CTX_set_params(3)>,
 L<EVP_KDF-derive(3)>,
 L<EVP_KDF-CTX(3)/PARAMETERS>
```
Comment (levitte) [inline doc/man7/EVP_KDF-TLS1_PRF.pod]: Same here

File: doc/man7/EVP_KDF-X942.pod [hunk 7]
```diff
@@ -99,9 +99,9 @@ RFC 2631
...
 
 L<EVP_KDF>,
-L<EVP_KDF-CTX_new_id(3)>,
+L<EVP_KDF-CTX_new(3)>,
 L<EVP_KDF-CTX_free(3)>,
-L<EVP_KDF-ctrl(3)>,
+L<EVP_KDF-CTX_set_params(3)>,
 L<EVP_KDF-size(3)>,
 L<EVP_KDF-derive(3)>,
 L<EVP_KDF(3)/PARAMETERS>
```
Comment (levitte) [inline doc/man7/EVP_KDF-X942.pod]: Same here

File: doc/man7/EVP_KDF-X963.pod [hunk 8]
```diff
@@ -88,9 +88,9 @@ value "label":
...
 
 L<EVP_KDF>,
-L<EVP_KDF-CTX_new_id(3)>,
+L<EVP_KDF-CTX_new(3)>,
 L<EVP_KDF-CTX_free(3)>,
-L<EVP_KDF-ctrl(3)>,
+L<EVP_KDF-CTX_set_params(3)>,
 L<EVP_KDF-size(3)>,
 L<EVP_KDF-derive(3)>,
 L<EVP_KDF-CTX(3)/PARAMETERS>
```
Comment (levitte) [inline doc/man7/EVP_KDF-X963.pod]: Same here

**Comment by paulidale** [comment]
All fixed.  Also picked up an inconsistency in the PARAMETERS section naming which I've also addressed.

**Comment by paulidale** [comment]
Ping?

File: doc/man7/EVP_KDF-HKDF.pod [hunk 9]
```diff
@@ -135,16 +135,16 @@ RFC 5869
...
-L<EVP_KDF-ctrl(3)>,
-L<EVP_KDF-size(3)>,
-L<EVP_KDF-derive(3)>,
-L<EVP_KDF-CTX(3)/PARAMETERS>
+L<EVP_KDF_CTX_new(3)>,
+L<EVP_KDF_CTX_free(3)>,
+L<EVP_KDF_size(3)>,
+L<EVP_KDF_CTX_set_params(3)>,
+L<EVP_KDF_derive(3)>,
+L<EVP_KDF/PARAMETERS>
```
Comment (levitte) [inline doc/man7/EVP_KDF-HKDF.pod]: `L<EVP_KDF(3)/PARAMETERS>`

Comment (paulidale) [reply doc/man7/EVP_KDF-HKDF.pod]: Fixed.

File: doc/man7/EVP_KDF-HKDF.pod [hunk 10]
```diff
@@ -135,16 +135,16 @@ RFC 5869
 =head1 SEE ALSO
 
 L<EVP_KDF>,
```
Comment (levitte) [inline doc/man7/EVP_KDF-HKDF.pod]: `L<EVP_KDF(3)>`

(needs to be fixed in multiple places)

Comment (paulidale) [reply doc/man7/EVP_KDF-HKDF.pod]: Fixed now.

There are some `B<EVP_KDF>` left but they are discussion the structure pointers.

**Comment by paulidale** [comment]
Merged, thanks for the feedback.

**Pull request closed without merging**